pkg/rc/replication_controller.go

```diff
@@ -0,0 +1,239 @@
+package rc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/rcstore"
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/util"
+)
+
+const (
+	rcIdLabel  = "replication_controller_id"
+	podIdLabel = "pod_id"
+)
+
+type ReplicationController interface {
+	ID() string
+	Manifest() *pods.Manifest
+	NodeSelector() labels.Selector
+	PodLabels() labels.Set
+	Disabled() bool
+	DesiredReplicas() int
+	CurrentNodes() ([]string, error)
+
+	// WatchDesires() spawns a goroutine that continually watches the desired replica count
+	// of this replication controller.
+	// Upon seeing any changes, it will converge observed state to desired state.
+	// The goroutine continues running until a value is received on the quitAndAck channel.
+	// The goroutine signals it has stopped by sending back a value on quitAndAck.
```

Comment:
same comment as before - closing the error channel would probably be sufficient to ack the quit request.